cmd/plugins/juju-metadata/toolsmetadata.go

```diff
@@ -62,23 +62,24 @@ use the --clean option.
 Examples:
 
 # generate metadata for "released":
-juju metadata generate-tools -d <workingdir>
+juju metadata generate-agents -d <workingdir>
 
 # generate metadata for "released":
-juju metadata generate-tools -d <workingdir> --stream released
+juju metadata generate-agents -d <workingdir> --stream released
```

Comment:
I think this all generally falls apart because of the RPN syntax, especially since we're mixing forward and reverse.
'generate-agent-metadata' sounds good, I haven't come up with names around this that sound particularly better using 'juju metadata ... ?'
juju metadata agent-generation
juju metadata generate-for-agents
juju metadata generate agent
I only like the one, but its not really possible to do.
This might be the best we got.

I wonder about

juju metadata generate-agent

leaving 'agent' there as intending to be an adjective.. ?

Reply:
Yes, the problem with 'metadata' is another one of our super commands but since this one is in 'plugins' directory, we kind of let it be...

Looking at the overall PR, I think I would have liked 'agent-binaries' but it feels too long for things like 'juju metadata generate-agent-binaries'... On the other hand, shortening it to just 'binaries' could be a bit ambiguous or too broad... 'juju metadata generate-binaries'...?...
One thing for sure, I am very happy to have put this change up, since I think the more eyes we have, the more likely we come up with a name that is valid/sound in all circumstances.

Reply:
we've decided to keep this as *-agents for now